engine/src/main/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionProcessor.java

```diff
@@ -130,6 +151,25 @@ private void deleteDecision(final PersistedDecision persistedDecision) {
     stateWriter.appendFollowUpEvent(keyGenerator.nextKey(), DecisionIntent.DELETED, decisionRecord);
   }
 
+  private void deleteProcess(final DeployedProcess process) {
+    // We don't add the checksum or resource in this event. The checksum is not easily available
+    // and the resources are left out to prevent exceeding the maximum batch size.
+    final var processRecord =
+        new ProcessRecord()
+            .setBpmnProcessId(process.getBpmnProcessId())
+            .setVersion(process.getVersion())
+            .setKey(process.getKey())
+            .setResourceName(process.getResourceName());
+    stateWriter.appendFollowUpEvent(keyGenerator.nextKey(), ProcessIntent.DELETING, processRecord);
+
+    final var runningInstances =
+        elementInstanceState.getProcessInstanceKeysByDefinitionKey(process.getKey());
```

Comment:
❌ This call has no possibility of being limited, so this processor can potentially block the stream processor for a really long time, causing the partition to go unhealthy and the leader stepping down. Any new leader would run into the same problem, effectively bringing down the partition altogether.

🔧 We could add a method that checks whether there are any process instances keys for this process definition key. That is a direct lookup, if there is an entry with prefix then true, otherwise false.